File: 1291-sequential-digits/1291-sequential-digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        string s = "123456789";
        int n = s.size(), left = 0, right = 0;
        
        vector<int> ans;
        for(int len = 2; len <= n; len++){
            left = 0;
            right = len - 1;
            while(right < n){
                int val = stoi(s.substr(left, right - left + 1));
                if(val < low){
                    left++;
                    right++;
                    continue;
                }
                if(val > high){
                    left++;
                    right++;
                    continue;
                }
                ans.push_back(val);
                left++;
                right++;
            }
        }

        // TC: O(1) — only 36 sequential digit numbers exist
        return ans;
    }
};

File: 1291-sequential-digits/1291-sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        s = "123456789"
        ans: List[int] = []
        for window in range(2, 10):
            left, right = 0, window-1
            while right < 9:
                val = int(s[left:right+1])
                #left, right = left + 1, right + 1
                left, right = left + 1, right + 1
                if val < low or val > high:
                    continue
                
                ans.append(val)

        return ans
        